tests/network/vmi_subdomain.go

```diff
@@ -155,7 +155,8 @@ func assertFQDNinGuest(vmi *v1.VirtualMachineInstance, expectedFQDN string) erro
 	return console.SafeExpectBatch(vmi, []expect.Batcher{
 		&expect.BSnd{S: "\n"},
 		&expect.BExp{R: console.PromptExpression},
-		&expect.BSnd{S: "hostname -f\n"},
+		// Delete 'myhostname' from nsswitch.conf as it prevents 'hostname -f' from getting FQDN.
+		&expect.BSnd{S: "sed -i -e 's/myhostname//g' /etc/nsswitch.conf && hostname -f\n"},
```

Comment:
Yep, /etc/nsswitch.conf in Fedora 35 looks a bit wierd. Just if you havent checked already: I put my findings on this in the PR description.

Reply:
Lets do this sed as part of the provision [0] ?
I wonder if it will affect the behavior of [1] as well

Note that for all the machines that have `CloudInitNoCloud` we do provide a hostname in the metadata of cloud-init (done internally by kubevirt [2])
so better to make sure that this feature is not broken by fedora 35 and masked by the change [1]
(however not all the tests use a fedora with cloud-init)
Changing as suggested on [0] would make the flow the same for all the tests, and more predictable imo

[0] https://github.com/kubevirt/kubevirtci/pull/713/files#diff-60e708f2e40cf0f5d49420a079fdd43d2a2b2691cada7e745ee1033d195d8a80R44
[1] https://github.com/kubevirt/kubevirt/pull/6958#discussion_r779701844
[2] https://github.com/kubevirt/kubevirt/blob/7c10ae70135b9262122feab31db2026d756f76b5/pkg/cloud-init/cloud-init.go#L361

Reply:
> Yep, /etc/nsswitch.conf in Fedora 35 looks a bit wierd. Just if you havent checked already: I put my findings on this in the PR description.

Well, we are suppose to take new distributions in early to confirm we can integrate well with them as guests. However, if the problems comes from the distribution, I am not sure if it is good for "us" to adjust to something wrong.
This specific tests is interested in the correct hostname to be use, so manipulating the guest feels wrong to me.
I guess, that is a good enough reason to leave this test working with the previous image.

However, maybe we can confirm this is indeed a bug by opening a BZ against Fedora and if this is confirmed we could patch it in the provision stage with a ref to the bug and an issue that tracks it (so we can revert the patch).

Not sure what is the next best action to be taken here (two images, or ref a BZ and patching the distribution).

Reply:
Diving a bit more into the topic... found the following: `man nss-resolve` [1] states:

> it is recommended to place "resolve" early in /etc/nsswitch.conf's "hosts:" line. It should be before the "files" entry, since systemd-resolved supports /etc/hosts internally, but with caching. To the contrary, it should be after "mymachines", to give hostnames given to local VMs and containers precedence over names received over DNS. Finally, we recommend placing "dns" somewhere after "resolve", to fall back to nss-dns if systemd-resolved.service is not available.

then `man nss-myhostname` [2]:

> It is recommended to place "myhostname" after "file" and before "dns". This resolves well-known hostnames like "localhost" and the machine hostnames locally. It is consistent with the behaviour of nss-resolve, and still allows overriding via /etc/hosts.

So the recommended **correct** `hosts:` line in /etc/nsswitch.conf should look like the following:

> mymachines resolve [!UNAVAIL=return] **files myhostname** dns

While in Fedora 35 it is:

> **files myhostname** resolve [!UNAVAIL=return] dns

So it seems that `resolve` is positioned wrong here. With `resolve [!UNAVAIL=return] **files myhostname** dns` fqdn works correctly.

[1] https://www.freedesktop.org/software/systemd/man/nss-resolve.html
[2] https://www.freedesktop.org/software/systemd/man/nss-myhostname.html

Reply:
Nice, i wonder just what /etc/hosts has, 
maybe it has a value that didn't exists on previous fedora, and it cause the order mess ?

Reply:
> i just wonder what /etc/hosts has

hm, nothing special IMHO:

```bash
[fedora@fc35 ~]$ cat /etc/hosts
127.0.0.1   localhost localhost.localdomain localhost4 localhost4.localdomain4
::1         localhost localhost.localdomain localhost6 localhost6.localdomain6
[fedora@fc35 ~]$ grep "hosts: " /etc/nsswitch.conf
hosts:      files myhostname resolve [!UNAVAIL=return] dns
[fedora@fc35 ~]$ hostname -f
fc35
[fedora@fc35 ~]$ sudo sed -i -e 's/myhostname//g' /etc/nsswitch.conf && hostname -f
fc35.testsubdomain.default.svc.cluster.local
```

Reply:
Thanks for checking

Reply:
> Thanks for checking

:ok_hand: 

... and now there is an open bug for systemd [1] along with proposed fix [2]

[1] https://github.com/systemd/systemd/issues/22081
[2] https://github.com/systemd/systemd/pull/22078

Reply:
Hey @EdDev , @oshoval , @AlonaKaplan , what do you think about this? I moved the patching of `/etc/nsswitch.conf` to the provision step [1] so no need to adjust the tests here anymore. There is already a BZ against fedora [2]. Other than that I do not observe any problems with the new image. After several rebases the test runs seem quite stable. Would it be fine to go forward with this PR or are you still unsure and have concerns about it?

[1] https://github.com/kubevirt/kubevirtci/pull/713
[2] https://bugzilla.redhat.com/show_bug.cgi?id=2038634

Reply:
I am fine with it, thanks, lets see other opinions if needed,
maybe worth to add a comment in the test as well  (referencing the change on kubevirtci, that has all the info) ?
so once it is fixed by fedora, in case it will breaks it will be visible easily in the test?
(not sure it will break the test, but if so)

Reply:
LGTM
@vasiliy-ul will the upstream fix break any test? In obvious ways?
@EdDev @AlonaKaplan any additional concern?

Reply:
>  @vasiliy-ul will the upstream fix break any test? In obvious ways?

If the fix for systemd is accepted it will affect only `nss-myhostame`. Cannot thinnk of any issue with that. Should work even with the patched `nsswitch.conf`. We just change the priority of the resolve modules there.